privacyidea/lib/security/aeshsm.py

```diff
@@ -314,6 +315,40 @@ def _decrypt_value(self, crypt_value, key_id):
 
         return self.decrypt(data, iv, key_id)
 
+    def create_keys(self):
+        """
+        Connect to the HSM and create the encryption keys.
+        The HSM connection should already be configured in pi.cfg.
+
+        We will create new keys with new key labels
+        :return: a dictionary of the created key labels
+        """
+        # We need a new read/write session
+        session = self.pkcs11.openSession(self.slot, PyKCS11.CKF_SERIAL_SESSION | PyKCS11.CKF_RW_SESSION)
+        session.login(self.password)
```

Comment:
I have no HSM to test :-) But as ``AESHardwareSecurityModule.__init__`` handles opening a session and logging in already, couldn't we just use ``self.session`` in this function?

Reply:
No, since the session that is already open is not a ``RW`` session. I am not sure, if we can actually ``WRITE`` new keys. The existing session is only opened to sign/decrypt things.

See ``CKF_RW_SESSION`` (which is not the default)

Reply:
You're right, I missed this.

Reply:
:+1: